add docs for dapr annotate command (#2576)
dapr/docs: daprdocs/content/en/reference/cli/dapr-annotate.md

| `--metrics-port` | | `-1` | The port to expose the metrics on |
| `--namespace, -n` | | | The namespace the resource target is in (can only be set if `--resource` is also set) |
| `--readiness-probe-delay` | | `-1` | The delay to use for the readiness probe in the sidecar. Read more [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes).|
| `--readiness-probe-period` | | `-1` | The period to use for the readiness probe in the sidecar. Read more [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes). |
| `--readiness-probe-threshold` | | `-1` | The threshold to use for the readiness probe in the sidecar. Read more [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes). |
| `--readiness-probe-timeout` | | `-1` | The timeout to use for the readiness probe in the sidecar. Read more [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes). |

> | `--resource, -r` | | | The resource to target to annotate |

Reviewer: TODO: Add other annotations from https://github.com/dapr/cli/issues/1018

Author: done

{{% alert title="Warning" color="warning" %}}

> If an application ID is not provided using `--app-id, -a`, we will generate one using the format `<namespace>-<kind>-<name>`.

Reviewer 2: I would have thought app-id is mandatory. And if not

```suggestion
If an application ID is not provided using `--app-id, -a`, and ID is generated using the format `<namespace>-<kind>-<name>`.
```

Reviewer 3: `app-id` is mandatory in the manifest but not in the CLI command. This is because you may be modifying an existing manifest that already has an app-id annotation - for instance `dapr annotate --log-level debug https://raw.githubusercontent.com/dapr/quickstarts/master/tutorials/hello-kubernetes/deploy/node.yaml` - this is just changing the log level of an existing dapr app so doesn't need to redefine `--app-id=nodeapp`. Rather than having to overwrite the app-id every time I just made it optional with a default convention if not provided by the CLI and not already present in the manifest. If you'd rather we didn't use this convention then I can make `app-id` flag required in the CLI. Let me know.

---
type: docs
title: "annotate CLI command reference"
linkTitle: "annotate"
description: "Add dapr annotatations to a Kubernetes configuration"
---

### Description

> Add dapr annotations to a Kubernetes configuration.

Reviewer 2: I presume arguments here reflects this list https://v1-8.docs.dapr.io/reference/arguments-annotations-overview/ 
If so has this list been checked since I could not find dapr.io/sidecar-image as a value.

Reviewer 2: If the above is true (the list) then I suggest the following update. 

```suggestion
Add Dapr annotations to a Kubernetes configuration. This enables you to change the Dapr annotations on a deployment other than those specified in the deployment file. See [Kubernetes annotations]({{< ref arguments-annotations-overview >}}) for a full description of each annotation.
```

Author: @msfussell I believe that this command as it exists is not for all the annotations as it exists. I am also thinking that all annotations should not be supported via this command. I am planning on creating a proposal to change this command to follow a helm based `--set` based format for 1.9. I will make note of the annotations needed to allowed to changed there.

| `--metrics-port` | | `-1` | The port to expose the metrics on |
| `--namespace, -n` | | | The namespace the resource target is in (can only be set if `--resource` is also set) |
| `--readiness-probe-delay` | | `-1` | The delay to use for the readiness probe in the sidecar. Read more [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes).|
| `--readiness-probe-period` | | `-1` | The period to use for the readiness probe in the sidecar. Read more [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes). |
| `--readiness-probe-threshold` | | `-1` | The threshold to use for the readiness probe in the sidecar. Read more [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes). |
| `--readiness-probe-timeout` | | `-1` | The timeout to use for the readiness probe in the sidecar. Read more [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes). |

> | `--resource, -r` | | | The resource to target to annotate |

Reviewer 2: What is --resource? When is this used?

Author: See examples with the `-r` flag ... It is the Kubernetes deployment resource that is the target for the annotate command ... 
```
# Annotate multiple deployments by name in a chain
kubectl get deploy -o yaml | dapr annotate -k -r nodeapp - | dapr annotate -k -r pythonapp - | kubectl apply -f -
```

More examples in the file ...
